packages/react-core/src/components/BackgroundImage/BackgroundImage.tsx

```diff
@@ -1,98 +1,70 @@
   className?: string;
+  /** Override svg filter to use */
+  filter?: React.ReactElement;
   /** Override image styles using a string or BackgroundImageSrc */
   src: string | BackgroundImageSrcMap;
 }
 
-export const BackgroundImage: React.FunctionComponent<BackgroundImageProps> = ({
-  className = '',
-  src,
-  ...props
-}: BackgroundImageProps) => {
-  let srcMap = src;
-  // Default string value to handle all sizes
-  if (typeof src === 'string') {
-    srcMap = {
-      [BackgroundImageSrc.xs]: src,
-      [BackgroundImageSrc.xs2x]: src,
-      [BackgroundImageSrc.sm]: src,
-      [BackgroundImageSrc.sm2x]: src,
-      [BackgroundImageSrc.lg]: src,
-      [BackgroundImageSrc.filter]: '' // unused
-    };
-  }
+let filterCounter = 0;
+
+export const BackgroundImage = ({ className, src, filter = defaultFilter, ...props }: BackgroundImageProps) => {
+  const getUrlValue = (size: keyof BackgroundImageSrcMap) => {
+    if (typeof src === 'string') {
+      return `url(${src})`;
+    } else if (typeof src === 'object') {
+      return `url(${src[size]})`;
+    }
 
-  // Build stylesheet string based on cssVariables
-  let cssSheet = '';
-  (Object.keys(cssVariables) as [keyof typeof srcMap]).forEach(size => {
-    cssSheet += `${cssVariables[size as keyof typeof cssVariables]}: url('${srcMap[size]}');`;
-  });
+    return '';
+  };
 
-  // Create emotion stylesheet to inject new css
-  const bgStyles = StyleSheet.create({
-    bgOverrides: `&.pf-c-background-image {
-      ${cssSheet}
-    }`
-  });
+  const filterId = `patternfly-background-image-filter-overlay${filterCounter++}`;
+  const style = {
+    [cssVar.name]: getUrlValue('xs'),
+    [cssVarName2x.name]: getUrlValue('xs2x'),
+    [cssVarNameSm.name]: getUrlValue('sm'),
+    [cssVarNameSm2x.name]: getUrlValue('sm2x'),
+    [cssVarNameLg.name]: getUrlValue('lg'),
+    [cssVarNameFilter.name]: `url(#${filterId})`
+  } as React.CSSProperties;
 
   return (
-    <div className={css(styles.backgroundImage, bgStyles.bgOverrides, className)} {...props}>
+    <div className={css(styles.backgroundImage, className)} style={style} {...props}>
       <svg xmlns="http://www.w3.org/2000/svg" className="pf-c-background-image__filter" width="0" height="0">
-        <filter id="image_overlay">
-          <feColorMatrix
-            type="matrix"
-            values="1 0 0 0 0
-            1 0 0 0 0
-            1 0 0 0 0
-            0 0 0 1 0"
-          />
-          <feComponentTransfer colorInterpolationFilters="sRGB" result="duotone">
-            <feFuncR type="table" tableValues="0.086274509803922 0.43921568627451" />
-            <feFuncG type="table" tableValues="0.086274509803922 0.43921568627451" />
-            <feFuncB type="table" tableValues="0.086274509803922 0.43921568627451" />
-            <feFuncA type="table" tableValues="0 1" />
-          </feComponentTransfer>
-        </filter>
+        {React.cloneElement(filter, { id: filterId })}
```

Comment:
why assign an id?

Reply:
To allow multiple background filters on the same page.